pkg/uninstaller/resources.go

```diff
@@ -0,0 +1,200 @@
+}
+
+type resultType struct {
+	Resource toCheckDeleted
+	Success  bool
+}
+
+var (
+	podGVR = schema.GroupVersionResource{
+		Group:    "",
+		Version:  "v1",
+		Resource: "pods",
+	}
+
+	toCheck = []toCheckDeleted{
+		{
+			gvr:           v1alpha1.TunnelEndpointGroupVersionResource,
+			labelSelector: metav1.LabelSelector{},
+		},
+		{
+			gvr:           v1alpha1.NetworkConfigGroupVersionResource,
+			labelSelector: metav1.LabelSelector{},
+		},
+		{
+			gvr: podGVR,
+			labelSelector: metav1.LabelSelector{
+				MatchLabels: map[string]string{"app": "virtual-kubelet"},
+			},
+		},
+		{
+			gvr: podGVR,
+			labelSelector: metav1.LabelSelector{
+				MatchLabels: map[string]string{"app": "broadcaster"},
+			},
+		},
+	}
+)
+
+func UnjoinClusters(client dynamic.Interface) error {
+	r1 := client.Resource(discoveryV1alpha1.ForeignClusterGroupVersionResource)
+	t, err := r1.Namespace("").List(context.TODO(), metav1.ListOptions{TypeMeta: metav1.TypeMeta{}})
+	if err != nil {
+		return err
+	}
+	klog.V(8).Info("Getting ForeignClusters list")
+	var foreign discoveryV1alpha1.ForeignClusterList
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(t.UnstructuredContent(), &foreign); err != nil {
+		return err
+	}
+	klog.V(8).Infof("Unjoin %v ForeignClusters", len(foreign.Items))
+	for _, item := range foreign.Items {
+		patch := []byte(`{"spec": {"join": false}}`)
+		_, err = r1.Namespace(item.Namespace).Patch(context.TODO(), item.Name, types.MergePatchType, patch, metav1.PatchOptions{})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func DisableBroadcasting(client dynamic.Interface) error {
+	r1 := client.Resource(clusterconfigV1alpha1.ClusterConfigGroupVersionResource)
+	t, err := r1.Namespace("").List(context.TODO(), metav1.ListOptions{TypeMeta: metav1.TypeMeta{}})
+	if err != nil {
+		return err
+	}
+	klog.V(8).Infof("Getting clusterConfigs")
+	var clusterconfigs clusterconfigV1alpha1.ClusterConfigList
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(t.UnstructuredContent(), &clusterconfigs); err != nil {
+		return err
+	}
+	klog.V(8).Infof("Patching ClusterConfigs")
+	for _, item := range clusterconfigs.Items {
+		patch := []byte(`{"spec": {"advertisementConfig": { "outgoingConfig" : { "enableBroadcaster" : false}}}}`)
+		_, err = r1.Namespace(item.Namespace).Patch(context.TODO(), item.Name, types.MergePatchType, patch, metav1.PatchOptions{})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func CheckReconciliation(client dynamic.Interface, toCheck toCheckDeleted) (bool, error) {
+	var (
+		objectList  *unstructured.UnstructuredList
+		err         error
+		labelString string
+	)
+
+	if labelString, err = generateLabelString(toCheck.labelSelector); err != nil {
+		return false, err
+	}
+	options := metav1.ListOptions{
+		LabelSelector: labelString,
+	}
+
+	objectList, err = client.Resource(toCheck.gvr).Namespace("").List(context.TODO(), options)
+
+	if apierrors.IsNotFound(err) {
```

Comment:
i think we do not need this check. List will return an UnstructuredList with the field Items having length 0 if no objects have been found.